modules/freewheelSSPBidAdapter.js

```diff
@@ -0,0 +1,303 @@
+    }
+  }
+  script += '};' +
+
+  'window.com.stickyadstv.' + getAPIName(bid.params.format) + '.start(config);';
+
+  return script;
+};
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: ['video'],
+  aliases: ['stickyadstv'], //  former name for freewheel-ssp
+  /**
+  * Determines whether or not the given bid request is valid.
+  *
+  * @param {object} bid The bid to validate.
+  * @return boolean True if this is a valid bid, and false otherwise.
+  */
+  isBidRequestValid: function(bid) {
+    return !!(bid.params.zoneId);
+  },
+
+  /**
+  * Make a server request from the list of BidRequests.
+  *
+  * @param {BidRequest[]} bidRequests A non-empty list of bid requests which should be sent to the Server.
+  * @return ServerRequest Info describing the request to the server.
+  */
+  buildRequests: function(bidRequests) {
+    // var currency = config.getConfig(currency);
+
+    this._currentBidRequest = bidRequests[0];
+    if (bidRequests.length > 1) {
+      console.log('Prebid.JS - freewheel bid adapter: only one ad unit is required.');
+    }
+
+    var requestParams = {
+      reqType: 'AdsSetup',
+      protocolVersion: '2.0',
+      zoneId: this._currentBidRequest.params.zoneId,
+      componentId: getComponentId(this._currentBidRequest.params.format)
+    };
+
+    var location = utils.getTopWindowUrl();
+    if (isValidUrl(location)) {
+      requestParams.loc = location;
+    }
+
+    this._currentPlayerSize = getBiggerSize(this._currentBidRequest.sizes);
+    if (this._currentPlayerSize[0] > 0 || this._currentPlayerSize[1] > 0) {
+      requestParams.playerSize = this._currentPlayerSize[0] + 'x' + this._currentPlayerSize[1];
+    }
+
+    return {
+      method: 'GET',
+      url: FREEWHEEL_ADSSETUP,
+      data: requestParams
+    };
+  },
+
+  /**
+  * Unpack the response from the server into a list of bids.
+  *
+  * @param {*} serverResponse A successful response from the server.
+  * @return {Bid[]} An array of bids which were nested inside the server.
+  */
+  interpretResponse: function(serverResponse) {
+    if (typeof serverResponse == 'object' && typeof serverResponse.body == 'string') {
+      serverResponse = serverResponse.body;
+    }
+
+    var xmlDoc;
+    try {
+      var parser = new DOMParser();
+      xmlDoc = parser.parseFromString(serverResponse, 'application/xml');
+    } catch (err) {
+      console.warn('Prebid.js - ' + BIDDER_CODE + ' : ' + err);
+      return;
+    }
+
+    const princingData = getPricing(xmlDoc);
+    const creativeId = getCreativeId(xmlDoc);
+
+    const topWin = getTopMostWindow();
+    if (!topWin.freeheelssp_cache) {
+      topWin.freeheelssp_cache = {};
+    }
+    topWin.freeheelssp_cache[this._currentBidRequest.adUnitCode] = serverResponse;
+
+    const bidResponses = [];
+
+    if (princingData.price) {
+      const bidResponse = {
+        requestId: this._currentBidRequest.bidId,
+        bidderCode: this._currentBidRequest.bidder,
```

Comment:
This isn't necessary as it is automatically added by the bidderFactory

Reply:
Removed.